spec/classes/pulpcore_spec.rb

```diff
@@ -20,6 +20,58 @@
         end
       end
 
+      context 'with worker_count set to 5' do
+        let(:params) { { worker_count: 5 } }
+
+        context 'with 6 workers from fact' do
+          let(:facts) { super().merge(pulpcore_workers: (1..6).map { |n| "pulpcore-worker@#{n}.service" } ) }
+
+          it 'is expected to reduce the workers to 5' do
+            is_expected.to compile.with_all_deps
+            is_expected.to contain_service('pulpcore-worker@5')
```

Comment:
@ekohl could you recommend syntax to actually loop over workers 1 through 5 here rather than just checking the fifth worker?

Reply:
In Ruby you can do it:
```ruby
it 'is expected to reduce the workers to 5' do
  (1..5).each do |n|
    is_expected.to contain_service("pulpcore-worker@#{n}")
  end
end
```

Reply:
Thanks, I wasn't sure about nesting it within `it...do`

Reply:
https://www.rubyguides.com/2016/02/ruby-procs-and-lambdas/ might be interesting to read. The signature is roughly `it(message = nil, &block)`. Once you know that, it feels more familiar.